File: packages/php/src/Errors/ConfigException.php
<?php

declare(strict_types=1);

namespace Tesote\Sdk\Errors;

use RuntimeException;

/**
 * Bad SDK config detected at construction. Raised before any HTTP call.
 *
 * Intentionally not a TesoteException: it predates the request lifecycle,
 * so the request-summary / attempts fields aren't meaningful.
 */
final class ConfigException extends RuntimeException
{
}

File: packages/csharp/src/Errors/ConfigException.cs
using System;

namespace Tesote.Sdk.Errors;

/// <summary>Bad SDK configuration; raised at construction.</summary>
public sealed class ConfigException : TesoteException
{
    /// <summary>Construct with a message only.</summary>
    public ConfigException(string message)
        : base(message, "CONFIG", 0, null, null, null, null, null, 0, null)
    {
    }

    /// <summary>Construct with a message and underlying cause.</summary>
    public ConfigException(string message, Exception cause)
        : base(message, "CONFIG", 0, null, null, null, null, null, 0, cause)
    {
    }
}
